### PR TITLE
Adds stinging nettles as a hydroponics plant

### PR DESCRIPTION
**Pull request by colossusqw**
[LABEL] [feature] [Hydroponics]

## About the PR 
This PR adds stinging nettles to botany, they can be acquired by hacking into the seed vendor and will periodically inject any mob within a 1 tile radius with histamine or any other chems they might have, similar to rafflesia, however in a much smaller range and without creating smoke.
image
image

## Why's this needed? 
This is another fun way for botany to spread chems around, like rafflesia, however without spreading miasma smoke everywhere. It could be used as a weapon or used as a quick omnizine dispenser by medbay if you don't mind a little histamine, in general, another thing for plant enthusiasts to mess with.

```changelog
(u)Hans
(*)Added stinging nettle to the list of plants botany can make! (More info in minor changes)
(+)Stinging nettle will periodically inject anyone close with histamine and the chemicals spliced with it, similar to rafflesia but without the smoke.
(+)Stinging nettle can be circumvented by wearing chemical protection, and the frequency of injecting people depends on it's endurance, while the amount of chemicals varies upon potency.
(+)Leaf sprite provided by Clover.
```

### CODE REVIEW

File: code/modules/hydroponics/plants_herb.dm [hunk 1]
```diff
@@ -209,6 +209,63 @@ ABSTRACT_TYPE(/datum/plant/herb)
...
+
+	HYPharvested_proc(var/obj/machinery/plantpot/POT,var/mob/user) //better not try to harvest these without gloves
+		..()
+		if (.) return
+		var/datum/plantgenes/DNA = POT.plantgenes
+		var/mob/living/carbon/human/H = user
+
+		if (H.hand)//gets active arm - left arm is 1, right arm is 0
+			if (istype(H.limbs.l_arm,/obj/item/parts/robot_parts) || istype(H.limbs.l_arm,/obj/item/parts/human_parts/arm/left/synth))
+				..()
```
Comment (TobleroneSwordfish) [inline code/modules/hydroponics/plants_herb.dm:254]: you've already called parent at the start of this proc, calling it twice may cause jank

File: code/obj/item/plants.dm [hunk 2]
```diff
@@ -343,6 +343,32 @@
 	icon_state = "mint"
 	brew_result = "menthol"
 
+/obj/item/plant/herb/nettle
+	name = "nettle leaves"
+	crop_suffix	= " leaves"
```
Comment (TobleroneSwordfish) [inline code/obj/item/plants.dm:348]:
```suggestion
	crop_suffix = " leaves"
```

**Review by TobleroneSwordfish — CHANGES_REQUESTED** [review]
Nettles :)

**Comment by Tarmunora** [comment]
Not really a fan but I suppose it can't be worse than rafflesia 😐

**Comment by ThatFiveGuys** [comment]
Honestly i think this plant fills too similar of a niche to rafflesia with less of the flavor of rafflesia that this would fit best as a procgen plant for the terrainifys

**Comment by nefarious6th** [comment]
find myself agreeing with ikea even though I liked this a lot initially. Would like to see it added at least for use in projects perhaps later. Would be a fun encounter on away zones, but my mind wraps back around to maybe having it generate in with histamine + random chem and then being able to uproot it and take with to re-plant.

Oh well. I think what I mean to say is I really like this and hope to see it in some aspect.

**Comment by Momoberry** [comment]
> Honestly i think this plant fills too similar of a niche to rafflesia with less of the flavor of rafflesia that this would fit best as a procgen plant for the terrainifys

It's openly more usable than rafflesia. Rafflesia mass spams out a mix of miasma and whatever chem you put in. The issue is that makes a wide area unusable for people that don't want to breathe in 300u of miasma lol. This is a much more contained plant, and honestly a lot more usable by botany for positive things.

**Comment by ThatFiveGuys** [comment]
> > Honestly i think this plant fills too similar of a niche to rafflesia with less of the flavor of rafflesia that this would fit best as a procgen plant for the terrainifys
> 
> It's openly more usable than rafflesia. Rafflesia mass spams out a mix of miasma and whatever chem you put in. The issue is that makes a wide area unusable for people that don't want to breathe in 300u of miasma lol. This is a much more contained plant, and honestly a lot more usable by botany for positive things.

The histamine makes it less usable for positive things (and also i have a pr in the works to address the miasma things for rafflesia)

**Comment by TobleroneSwordfish** [comment]
Maybe give it a gene strain or mutation called "dead nettles" that removes the histamine?

**Comment by ZWRh3** [comment]
i support this but i agree with TobleroneSwordfish that there should be a histamineless mutation i also have one concern

"gib nettles": water and potassium splices are iirc non explosive until they get transfered into something else eg. a joint or in this case whatever poor sap is close to the plant i dont know if it would do the ok thing (explode once destroying itself in the process too like some botany proximity bomb) or the bad thing stunlock with explosions until they gib

File: code/obj/item/plants.dm (hunk 2 above)
Comment (colossusqw) [reply code/obj/item/plants.dm:348]: huh, it's correct on my end, weird.

File: code/modules/hydroponics/plants_herb.dm [hunk 3]
```diff
@@ -209,6 +209,63 @@ ABSTRACT_TYPE(/datum/plant/herb)
...
+			if (istype(H.limbs.r_arm,/obj/item/parts/robot_parts) || istype(H.limbs.r_arm,/obj/item/parts/human_parts/arm/right/synth))
+				..()
+				return
+		if(istype(H))
+			if(H.gloves)
+				..()
+				return
+		boutput(user, "<span class='alert'>Your hands itch from touching [POT]!</span>")
+		for (var/plantReagent in assoc_reagents)
+			H.reagents?.add_reagent(plantReagent, 5 * round(max(1,(1 + DNA.potency / (10 * (length(assoc_reagents) ** 0.5))))))
```
[the remark this answers is not in the document]
Comment (colossusqw) [reply code/modules/hydroponics/plants_herb.dm:264]: the formula is not the same, it's similar but one of them uses your current chem protection, i could try to make a sting thing with a check for that though

**Comment by colossusqw** [comment]
Also, the histamine values have been changed for it to be a standard 5 units per time it injects you, making it be way less of a problem for non-evil injections

**Review by pali6 — APPROVED** [review]
Yeah sure, I've wanted something similar for a while.
Though my original idea was a grass mutation that injects you as you walk on the floor where the grass is growing.

**Pull request merged**